cliquet/scripts/cliquet.py

```diff
@@ -15,6 +16,12 @@ def deprecated_init(env):
 
 def init_schema(env):
     registry = env['registry']
+    settings = registry.settings
+
+    if settings.get('read_only', False):
+        message = 'You cannot migrate the database with a read_only setup.'
+        print(message, file=sys.stderr)
+        sys.exit(1)
```

Comment:
Side note:  the cache backend requires some write access, but we expect it to have been migrated with the two other backends